packages/@react-spectrum/s2/src/Button.tsx

```diff
@@ -45,14 +45,18 @@ interface ButtonStyleProps {
   staticColor?: 'white' | 'black'
 }
 
-export interface ButtonProps extends Omit<RACButtonProps, 'className' | 'style' | 'children' | 'onHover' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, StyleProps, ButtonStyleProps {
+export interface ButtonProps extends Omit<RACButtonProps, 'className' | 'style' | 'children' | 'onHover' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, UnsafeStyles, ButtonStyleProps {
   /** The content to display in the Button. */
-  children?: ReactNode
+  children?: ReactNode,
+  /** Spectrum-defined styles, returned by the `style()` macro. */
+  styles?: StylesPropWithHeight | StylesProp
```

Comment:
Why do we need to allow overriding the height of a button?

Reply:
parity with S1, I didn't find anything anywhere that indicated if that shouldn't be supported but happy to remove if we think it would be odd to customize the height of the button

Reply:
Was button height support because we didn't prevent style props for changing it or spectrum wants the button height to change?

Reply:
not sure, I've done a bunch of searches but its been hard to find anything about in slack

Reply:
Asked spectrum and they want to allow support for this so long as we aren't explicitly recommending teams go off and customize this